### DISCUSSION TITLE
wasm32-unknown-emscripten fall back

### DISCUSSION THREAD

**Discussion by JosiahParry** [rayon-rs/rayon#1155] [Q&A]
I have read that rayon has a single threaded fallback for wasm32-unknown-unknown. However, I ran into an issue with compiling rayon to the wasm32-unknown-emscripten target. Is there a fallback for this target?

**Comment by cuviper** [comment]
Please provide details -- what issue did you encounter?

The fallback does not look at the particular target, just that creating a thread returns [`io::ErrorKind::Unsupported`](https://doc.rust-lang.org/std/io/enum.ErrorKind.html#variant.Unsupported). If it fails for any other reason, rayon will pass that error through.
